massa-execution/src/interface_impl.rs

```diff
@@ -230,16 +230,35 @@ impl Interface for InterfaceImpl {
     }
 
     fn generate_event(&self, data: String) -> Result<()> {
-        let context = context_guard!(self);
+        let mut context = context_guard!(self);
         let slot = context.slot;
         let block = context.opt_block_id;
         let call_stack = context.call_stack.clone();
+        let read_only = context.read_only;
+        if !read_only {
+            context.created_event_index += 1;
+        }
+
+        let mut to_hash: Vec<u8> = slot.to_bytes_key().to_vec();
+        to_hash.append(&mut context.created_event_index.to_be_bytes().to_vec());
+        if read_only {
+            to_hash.push(0u8);
+        } else {
+            to_hash.push(1u8);
+        }
+
         let context = EventExecutionContext {
```

Comment:
keep the context lock to be able to push the event into it, don't override it with this variable